src/components/Tooltip/hooks/useTooltip.tsx

```diff
@@ -0,0 +1,58 @@
+import { isNotNull } from 'ramda-adjunct';
+import React, { useEffect, useRef, useState } from 'react';
+import usePortal from 'react-cool-portal';
+
+import { useCalculatePortaPlacement } from '../../../hooks/useCalculatePortalPlacement';
+import { StyleProps } from '../../../hooks/useCalculatePortalPlacement.types';
+import TooltipPopup from '../TooltipPopup';
+import { UseTooltipOptions, UseTooltipReturnType } from './useTooltip.types';
+
+const defaultTooltipPopupDimensions = { width: 270, space: 8 };
+
+export const useTooltip = (
+  parentRef: React.MutableRefObject<HTMLSpanElement>,
+  { defaultIsPopupDisplayed = false, placement }: UseTooltipOptions,
+): UseTooltipReturnType => {
+  const { Portal, isShow, show, hide } = usePortal({
+    defaultShow: defaultIsPopupDisplayed,
+  });
+  const [style, setStyle] = useState<StyleProps>(defaultTooltipPopupDimensions);
+  const getPlacementStyles = useCalculatePortaPlacement(parentRef, {
+    placement,
+    ...defaultTooltipPopupDimensions,
+  });
+  const scrollListener = useRef(null);
+  const isListenersAdded = useRef(false);
+
+  useEffect(() => {
+    setStyle(getPlacementStyles());
+  }, [getPlacementStyles, isShow]);
+
+  useEffect(() => {
+    scrollListener.current = () => {
+      hide();
+    };
+    if (isListenersAdded.current && isNotNull(scrollListener.current)) {
```

Comment:
Pls, just check `isNotNull(scrollListener.current)` is not enough? Wondering if `isListenersAdded` ref is needed or what use case I'm missing

Reply:
The problem is we want to remove event listener from window, before we add another one, but if you try to remove event listener which doesn't exist it throws. So this we we check if there is any previous listener which we can remove, if not we just skip the removing part

Reply:
Check of previous listener, got it